swagger2.0/oic.sec.sdi.swagger.json

```diff
@@ -0,0 +1,171 @@
+        },
+        "rt": {
+          "description": "Resource Type of the Resource.",
+          "items": {
+            "maxLength": 64,
+            "type": "string",
+            "enum": ["oic.r.sdi"]
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        },
+        "n": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/n"
+        },
+        "id": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/id"
+        },
+        "priv": {
+          "description": "Flag to indicate whether the Security Domain Information is copied to “/oic/res”, and thus, whether it is publicly visible or private.",
+          "type": "boolean"
+        },
+        "if" : {
+          "description": "The interface set supported by this Resource.",
+          "items": {
+            "enum": [ "oic.if.baseline", "oic.if.rw" ],
+            "type": "string"
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        }
+      },
+      "type" : "object",
+      "required": [ "uuid", "name", "priv" ]
+    },
+    
+    "Sdi-Update" : {
+      "properties": {
+        "uuid": {
+          "description": "A UUID that identifies the Security Domain, set by DOTS during onboarding.",
+          "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+          "type": "string"
+        },
+        "name": {
+          "description": "Human-friendly name for the Security Domain, set by DOTS during onboarding.",
+          "type": "string"
+        },
+        "rt": {
+          "description": "Resource Type of the Resource.",
+          "items": {
+            "maxLength": 64,
+            "type": "string",
+            "enum": ["oic.r.sdi"]
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        },
+        "n": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/n"
+        },
+        "id": {
+          "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/id"
+        },
+        "priv": {
+          "description": "Flag to indicate whether the Security Domain Information is copied to “/oic/res”, and thus, whether it is publicly visible or private.",
+          "type": "boolean"
+        },
+        "if" : {
+          "description": "The interface set supported by this Resource.",
+          "items": {
+            "enum": [ "oic.if.baseline", "oic.if.rw" ],
+            "type": "string"
+          },
+          "minItems": 1,
+          "readOnly": true,
+          "type": "array"
+        }
+      },
+      "type" : "object",
+      "required": [ "uuid", "name", "priv" ]
```

Comment:
This means that an UPDATE request must include all three Properties.  In other words, if you only want to set/change the "name", the UPDATE request has to also include "uuid" and "priv".  Is that the expected behavior?

Reply:
At least in RFOTM all 3 have to be provided by OBT in one message. In RFPRO, if OBT wants to reprovision name and priv, we expect it to include uuid as a safety mechanism (e.g. if OBT manages multiple domains, only /oic/sec/sdi with same uuid will update the name)